vortex-expr/src/scope.rs

```diff
@@ -6,18 +6,49 @@ use vortex_array::{Array, ArrayRef};
 use vortex_dtype::DType;
 use vortex_error::{VortexResult, vortex_err};
 
-use crate::IDENTITY_IDENTIFIER;
-
-pub type Identifier = Arc<str>;
 type ExprScope<T> = HashMap<Identifier, T>;
 
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
+pub enum Identifier {
+    Identity,
+    Other(Arc<str>),
+}
+
+impl From<String> for Identifier {
+    fn from(value: String) -> Self {
+        Self::Other(value.into())
```

Comment:
this might clash with ""

Reply:
I'll change that into `TryFrom` or `FromStr` and not allow the empty string